presto-main/src/main/java/com/facebook/presto/server/security/LdapFilter.java

```diff
@@ -0,0 +1,234 @@
+            return;
+        }
+
+        HttpServletRequest request = (HttpServletRequest) servletRequest;
+        HttpServletResponse response = (HttpServletResponse) servletResponse;
+        String header = request.getHeader(HttpHeaders.AUTHORIZATION);
+
+        if (header != null) {
+            List<String> parts = Splitter.onPattern("\\s+").splitToList(header);
+
+            checkState(parts.size() == 2, "Invalid value for authorization header");
+            checkState(parts.get(0).equalsIgnoreCase(AUTHENTICATION_TYPE), "Incorrect authentication type. Should be basic");
+
+            List<String> credentials = Splitter.on(":").splitToList(new String(Base64.getDecoder().decode(parts.get(1))));
+            checkState(credentials.size() == 2, "Username/Password missing in the request header");
+
+            DirContext context = null;
+            try {
+                String user = credentials.get(0);
+                String password = credentials.get(1);
+
+                Hashtable<String, String> environment = getBasicEnvironment();
+                String principal = getPrincipal(user);
+
+                environment.put(SECURITY_AUTHENTICATION, "simple");
+                environment.put(SECURITY_PRINCIPAL, principal);
+                environment.put(SECURITY_CREDENTIALS, password);
+                context = authenticate(environment);
+
+                //TODO: Use the context to do the group membership search
+
+                // ldap authentication ok, continue
+                nextFilter.doFilter(new HttpServletRequestWrapper(request)
+                {
+                    @Override
+                    public Principal getUserPrincipal()
+                    {
+                        return new LdapPrincipal(user);
+                    }
+                }, servletResponse);
+                return;
+            }
+            catch (AuthenticationException e) {
+                // LOG incorrect credentials message separately since the one from e is not explicit
+                LOG.debug("Invalid credentials. Check your username/password");
+                throw Throwables.propagate(e);
+            }
+            catch (NamingException e) {
+                throw Throwables.propagate(e);
+            }
+            finally {
+                closeQuietly(context);
+            }
+        }
+
+        // request for user/password for LDAP authentication
+        sendChallenge(response);
+    }
+
+    private Hashtable<String, String> getBasicEnvironment()
+    {
+        Hashtable<String, String> environment = new Hashtable<>();
+        environment.put(INITIAL_CONTEXT_FACTORY, LDAP_CONTEXT_FACTORY);
+        environment.put(PROVIDER_URL, ldapUrl);
+        return environment;
+    }
+
+    /**
+     * Returns the prinicipal which will be used to bind to LDAP
+     *
+     * @param user
+     * @return
+     * @throws NamingException
+     */
+    abstract String getPrincipal(String user) throws NamingException;
```

Comment:
Use delegation rather than inheritance. This class should take an interface for this.

It should probably throw `Exception` or nothing, since the implementation could be anything.